### PR TITLE
Update Jetpack Tracks Client library reference to not have a hardcoded directory path.

### PR DESCRIPTION
**Pull request by oskosk**
This PR updates the way the path to the Jetpack client library is forged so to take in account Jetpack installations that do not live in `plugins/jetpack`. It's the case for the Jetpack Beta Tester plugin, that it will use the `plugins/jetpack-dev` directory when running a development branch of Jetpack.

Note the reference to both `jetpack-dev` and `jetpack` in the directory in this error message:
```
Fatal error: Cannot redeclare jetpack_tracks_record_event_raw() (previously declared in wp-content/plugins/jetpack-dev/_inc/lib/tracks/client.php:29) in wp-content/plugins/jetpack/_inc/lib/tracks/client.php on line 31
```

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/ woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Updates  `class-wc-tracks-client.php` to rely on Jetpack's function `jetpack_require_lib_dir()` to figure out the base directory for Jetpack and forging the path to the `tracks/client.php` library.

### How to test the changes in this Pull Request:

1. Start with a site with WooCommerce, [Jetpack Beta Tester plugin](https://jetpack.com/download-jetpack-beta/), and `WP_DEBUG`, `WP_DEBUG_LOG` set to true. ([JN link](https://jurassic.ninja/create?jetpack-beta&woocommerce&wp-debug-log)).
2. Tail the log file. `tail -F wp-content/debug.log`. 
3. Run the setup wizard until the Payment Step. 
4. When clicking there, expect to reach the Shipping step without any error logged.

Before this fix, a fatal error was logged and shown in the screen.

image

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed compatibility with the Jetpack Beta Tester plugin when loading the Tracks client library.

### CODE REVIEW

**Comment by oskosk** [comment]
There's a merge commit shown in this PR. Sorry for that. It should go away if this PR is squashed when merged.

File: includes/tracks/class-wc-tracks-client.php [hunk 1]
```diff
@@ -137,10 +137,10 @@ public static function build_timestamp() {
 	 * @return array Identity properties.
 	 */
 	public static function get_identity( $user_id ) {
-		$jetpack_lib = trailingslashit( WP_PLUGIN_DIR ) . 'jetpack/_inc/lib/tracks/client.php';
+		$jetpack_lib = '_inc/lib/tracks/client.php';
 
-		if ( class_exists( 'Jetpack' ) && file_exists( $jetpack_lib ) ) {
-			include_once $jetpack_lib;
+		if ( class_exists( 'Jetpack' ) && file_exists( JETPACK__PLUGIN_DIR . $jetpack_lib ) ) {
```
Comment (enejb) [inline includes/tracks/class-wc-tracks-client.php]: Why even make the dependency on `class_exists( 'Jetpack' )`.

`is_defined( 'JETPACK__PLUGIN_DIR')` should be enough?

Comment (oskosk) [reply includes/tracks/class-wc-tracks-client.php]: I'd prefer to leave this exact pattern for checking if Jetpack is available in ths codebase for consistency. There used to be many file based checks before. `is_plugin_active( 'jetpack/jetpack.php' )`.

The preferred way for asserting that Jetpack is active on a site is `class_exists( 'Jetpack' )`.

**Review by kloon — APPROVED** [review]
LGTM

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Since we are changing it, probably better use `jetpack_require_lib_dir()` that allows Jetpack one day change the lib directory without breaking us:

https://github.com/Automattic/jetpack/blob/997b2c1bbe50e18cace4b069bbfe6890dfe7d566/jetpack.php#L45-L57

Also we have sure that this has been introduced in 4.0.2, if we need to declare any support for early version of Jetpack.

**Comment by oskosk** [comment]
Thanks @claudiosanches! Addressed your feedback in 120fdb2028aa8194b41a359c26d49a8bf727d7a7

**Review by claudiosanches — APPROVED** [review]
Thanks!

**Pull request merged**